### ISSUE TITLE
Relatively High Lag for Query Results

### ISSUE THREAD

**Issue by stefanseltmann77** [snowflakedb/snowflake-sqlalchemy#111]
Hello,
I’m currently using the connector in a project that involves a web-gui based on flask. 
In this application I’m experiencing lags from 150ms to 400ms or even more for each query passed along to the snowflake.
Session pooling is activated by default via the snowflake connector as far as I know. That's also what the log shows.
The mentioned lags for each query makes the connection to slow for a web UI

Setup:
- Python 3.6
- Linux-3.10.0-862.14.4.el7.x86_64-x86_64-with-centos-7.5.1804-Core
- SQLAlchemy  1.3.1
- snowflake-sqlalchemy  1.1.12 
- snowflake-connector-python  1.7.10 

Is there anything that can be done to decrease the lag?
Examples for this lag are in the lines:  531, 630, 658, 697, 731, 769 ... 
basically most of the lines including "connectionpool"

Thanks in advance!

[python_connector.log](https://github.com/snowflakedb/snowflake-sqlalchemy/files/3508824/python_connector.log)

**Comment by smtakeda** [comment]
Looks like those lags are reasonable to execute a query and get the results:
```
2019-08-16 09:30:10,571 - Thread-6 network.py:751 - _request_exec() - DEBUG - socket timeout: 60
2019-08-16 09:30:11,168 - Thread-6 connectionpool.py:383 - _make_request() - DEBUG - "POST /queries/v1/query-request?requestId=c390cc4d-d11b-4c23-b76d-5b4f46293680&request_guid=def43c8b-d6a9-45bb-aa58-70b5ff093160 HTTP/1.1" 200 11966
```
It includes the network latency from the client to Snowflake endpoint, SQL compile, execute, stage the results and back to the client.

**Comment by stefanseltmann77** [comment]
Ok, so it means, I have done nothing wrong, and I rather have to lower my expectations right?
Cheers,
Stefan

**Comment by villebro** [comment]
I think this is to be expected (OLAP vs OLTP). I think your use case might be more suited for something like Redis or Postgres.
